app/controllers/concerns/billable_event_trackable.rb

```diff
@@ -0,0 +1,54 @@
+module BillableEventTrackable
+  extend ActiveSupport::Concern
+
+  def track_billing_events
+    if current_session_has_been_billed?
+      update_sp_return_log(billable: false)
+    else
+      increment_sp_monthly_auths
+      update_sp_return_log(billable: true)
+      mark_current_session_billed
+      add_sp_cost(:authentication)
+    end
+  end
+
+  private
+
+  def increment_sp_monthly_auths
+    issuer = sp_session[:issuer]
+    MonthlySpAuthCount.increment(current_user.id, issuer, sp_session_ial)
```

Comment:
we can inline this assignment, and also this method seems like another good candidate to switch to keyword args?

```suggestion
    MonthlySpAuthCount.increment(user_id: current_user.id, issuer: sp_session[:issuer], ial: sp_session_ial)
```